SingularityService/src/main/java/com/hubspot/singularity/data/SingularityValidator.java

```diff
@@ -230,12 +230,32 @@ public SingularityRequest checkSingularityRequest(
       !request.getInstances().isPresent() || request.getInstances().get() > 0,
       "Instances must be greater than 0"
     );
-    checkBadRequest(
-      request.getInstancesSafe() <= maxInstancesPerRequest,
-      "Instances (%s) be greater than %s (maxInstancesPerRequest in mesos configuration)",
-      request.getInstancesSafe(),
-      maxInstancesPerRequest
-    );
+
+    // requested instances cannot exceed request-level max scale (if present) or global max scale
+    if (request.getMaxScale().isPresent()) {
+      if (request.getMaxScale().get() < maxInstancesPerRequest) {
+        checkBadRequest(
+          request.getInstancesSafe() <= request.getMaxScale().get(),
+          "Instances (%s) cannot be greater than %s (maxScale in request)",
+          request.getInstancesSafe(),
+          request.getMaxScale()
+        );
+      } else {
+        checkBadRequest(
+          request.getInstancesSafe() <= maxInstancesPerRequest,
+          "Instances (%s) cannot be greater than %s (maxInstancesPerRequest in mesos configuration)",
+          request.getInstancesSafe(),
+          maxInstancesPerRequest
+        );
+      }
+    } else {
+      checkBadRequest(
+        request.getInstancesSafe() <= maxInstancesPerRequest,
+        "Instances (%s) cannot be greater than %s (maxInstancesPerRequest in mesos configuration)",
+        request.getInstancesSafe(),
+        maxInstancesPerRequest
+      );
+    }
```

Comment:
may have been staring at this for too long, but I couldn't figure out a cleaner way of implementing this logic such that users would be told specifically which max scale config they were violating without being hacky